### ISSUE TITLE
sqlite WAL mode

### ISSUE THREAD

**Issue by dholth** [reclosedev/requests-cache#564]
### Feature Request

Make it easy to issue PRAGMA journal_mode=wal; when cache is created

### Use case

sqlite's WAL mode means writers do not block readers. This could help parallelism / prevent timeouts when many processes are trying to use the cache.

Once the pragma has been issued the journal mode is saved in the database.

Needs a filesystem that supports shared memory, afaik does not work on a network filesystem.

**Comment by JWCook** [comment]
Currently, you can run any pragma statements you want from `SQLiteDict.connection()`:
```python
from requests_cache import CachedSession

session = CachedSession()
with session.cache.responses.connection() as conn:
    conn.execute('PRAGMA journal_mode=wal;')
```

I would be fine with adding another keyword argument to enable WAL mode. Would you like to submit a PR for that?

**Comment by JWCook** [comment]
Closed by #565
